src/plasmapy/analysis/swept_langmuir/helpers.py

```diff
@@ -145,3 +155,82 @@ def check_sweep(  # noqa: C901, PLR0912
         current = current.value
 
     return voltage, current
+
+
+def sort_sweep_arrays(
+    voltage: np.ndarray,
+    current: np.ndarray,
+    voltage_order: Literal["ascending", "descending"] = "ascending",
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Sort the swept langmuir ``voltage`` and ``current`` traces to
+    ensure the ``voltage`` array is either monotonically increasing or
+    decreasing.
+
+    Parameters
+    ----------
+    voltage: `numpy.ndarray`
+        1D `numpy.ndarray` representing the voltage of the swept
+        Langmuir trace.  *No units are assumed or checked, but values
+        should be in volts.*
+
+    current: `numpy.ndarray`
+        1D `numpy.ndarray` representing the current of the swept
+        Langmuir trace.  Values should start from a negative
+        ion-saturation current and increase to a positive
+        electron-saturation current.  *No units are assumed or checked,
+        but values should be in amperes.*
+
+    voltage_order: `str`
+        Either 'ascending' or 'descending' to indicate how the
+        ``voltage`` array should be sorted.  (DEFAULT: ``'ascending'``)
+
+    Returns
+    -------
+    voltage : `numpy.ndarray`
+        Sorted ``voltage`` array.
+
+    current : `numpy.ndarray`
+        Matched ``current`` array to the sorted ``voltage`` array.
+    """
+    if not isinstance(voltage_order, str):
+        raise TypeError(
+            "Expected 'voltage_order' to be a string equal to 'ascending' "
+            f"or 'descending', but got type {type(voltage_order)}."
+        )
+    elif voltage_order not in ["ascending", "descending"]:
+        raise ValueError(
+            "Expected 'voltage_order' to be a string equal to 'ascending' "
+            f"or 'descending', but got '{voltage_order}'."
+        )
+
+    voltage, current = check_sweep(
+        voltage, current, strip_units=True, allow_unsorted=True
+    )
+
+    # determine order
+    voltage_diff = np.diff(voltage)
+    if np.all(voltage_diff >= 0):
+        _order = "ascending"
+    elif np.all(voltage_diff <= 0):
+        _order = "descending"
```

Comment:
This part might not work with a `Quantity` 🤔

Reply:
At this point `voltage` and `current` have passed through `check_sweep()` which strips units and returns them as `numpy` arrays.